unblob/finder.py

```diff
@@ -19,8 +28,67 @@
 """
 
 
+def search_chunks_by_priority(  # noqa: C901
+    path: Path, file: io.BufferedReader, file_size: int
+) -> List[ValidChunk]:
+    """Search all ValidChunks within the file.
+    Collect all the registered handlers by priority, search for YARA patterns and run
+    Handler.calculate_chunk() on the found matches.
+    We don't deal with offset within already found ValidChunks and invalid chunks are thrown away.
+    """
+    all_chunks = []
+
+    for priority_level, handlers in enumerate(_ALL_MODULES_BY_PRIORITY, start=1):
+        logger.info("Starting priority level", priority_level=noformat(priority_level))
+        yara_rules = make_yara_rules(tuple(handlers.values()))
+        yara_results = search_yara_patterns(yara_rules, handlers, path)
```

Comment:
Just so I won't forget: We make multiple passes over an input file in the hope that the first priority level would match most of the time. This assumption only works in the first few layers, as we are recursing into the contents eventually the opposite will happen. It would make more sense to me to aggregate all rules, do one paths and process the results according to priority.

Reply:
I added this case for the performance measurements:
https://github.com/IoT-Inspector/unblob/issues/100